base/file.jl

```diff
@@ -540,16 +541,30 @@ end # os-test
 
 
 """
-    tempname()
+    tempname(parent=tempdir(); cleanup=true) -> String
 
 Generate a temporary file path. This function only returns a path; no file is
-created. The path is likely to be unique, but this cannot be guaranteed.
+created. The path is likely to be unique, but this cannot be guaranteed due to
+the very remote posibility of two simultaneous calls to `tempname` generating
+the same file name. The name is guaranteed to differ from all files already
+existing at the time of the call to `tempname`.
+
+When called with no arguments, the temporary name will be an absolute path to a
+temporary name in the system temporary directory as given by `tempdir()`. If a
+`parent` directory argument is given, the temporary path will be in that
+directory instead. The `cleanup` option controls whether the temporary file is
+automatically deleted when the process exits.
```

Comment:
Can we clarify that anything at the location given by `tempname` will be deleting *if* something is created at that location.
Since `tempname` itself does not create a file.

But express that idea better that I did since.